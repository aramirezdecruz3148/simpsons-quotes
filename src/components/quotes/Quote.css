@import url('https://fonts.googleapis.com/css?family=Scope+One&display=swap');
@import url('https://fonts.googleapis.com/css?family=Luckiest+Guy&display=swap');

.section {
  height: 30vw;
  width: 40vw;
  text-align: center;
  margin: 0 3vw 3vw 0;
  padding-top: 3vw;
  border: .2vw solid black;
  border-radius: 10%;
  font-size: 1.4vw;
  font-weight: bolder;
  overflow: scroll;
  padding: 2vw;
}

.section img {
  width: 10vw;
  margin-bottom: 1vw;
}

.section h3 {
  font-family: 'Luckiest Guy', cursive;
  color: #FFD90F;
  font-size: 2.8vw;
  margin-top: -1vw;
  text-shadow:
		-.2vw -.2vw 0 #000,
		.2vw -.2vw 0 #000,
		-.2vw .2vw 0 #000,
		.2vw .2vw 0 #000;
}

.section p {
  text-align: center;
  font-weight: normal;
  font-style: italic;
  font-family: 'Scope One', serif;
  font-size: 1.5vw;
}

.section p:before {
  content: open-quote;
}
.section p:after {
  content: close-quote;
}