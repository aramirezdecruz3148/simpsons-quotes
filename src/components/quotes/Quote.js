import React from 'react';
import PropTypes from 'prop-types';

function Quote({ characterImage, characterName, quote }) {
  return (
    <section>
      <h3>{characterName}</h3>
      <img src={characterImage} alt={characterName} />
      <pre>{quote}</pre>
    </section>
  );
}

Quote.propTypes = {
  characterName: PropTypes.string.isRequired,
  characterImage: PropTypes.string.isRequired,
  quote: PropTypes.string.isRequired
};

export default Quote;
