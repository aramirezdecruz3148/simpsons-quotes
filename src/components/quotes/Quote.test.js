import React from 'react';
import { shallow } from 'enzyme';
import Quote from './Quote';

describe('Quote component', () => {
  it('renders Quote', () => {
    const wrapper = shallow(<Quote characterName={'name'} characterImage={'url'} quote={'wit'}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
