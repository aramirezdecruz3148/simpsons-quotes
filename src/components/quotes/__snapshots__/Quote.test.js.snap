// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote component renders Quote 1`] = `
<section
  className="section"
>
  <h3>
    name
  </h3>
  <img
    alt="name"
    src="url"
  />
  <p>
    wit
  </p>
</section>
`;
